src/plugins/data/server/search/types.ts

```diff
@@ -19,22 +19,27 @@
 
 import { RequestHandlerContext } from '../../../../core/server';
 import { IKibanaSearchResponse, IKibanaSearchRequest } from '../../common/search';
-import { ES_SEARCH_STRATEGY, IEsSearchRequest, IEsSearchResponse } from './es_search';
+import { IEsSearchRequest, IEsSearchResponse } from './es_search';
 
 export interface ISearchSetup {
   /**
    * Extension point exposed for other plugins to register their own search
    * strategies.
    */
-  registerSearchStrategy: TRegisterSearchStrategy;
+  registerSearchStrategy: (name: string, strategy: ISearchStrategy) => void;
 }
 
 export interface ISearchStart {
   /**
    * Get other registered search strategies. For example, if a new strategy needs to use the
    * already-registered ES search strategy, it can use this function to accomplish that.
    */
-  getSearchStrategy: TGetSearchStrategy;
+  getSearchStrategy: (name: string) => ISearchStrategy;
+  search: (
+    context: RequestHandlerContext,
+    request: IKibanaSearchRequest,
+    options: any
```

Comment:
Shouldn't this be

```suggestion
    options: ISearchOptions
```

Reply:
I just realized why it's not... because you're sending the `strategy` parameter here. I would rather see that as maybe a new parameter passed to the function (we could add it to both the client/server side signatures and have it default to `"es"`).